### ISSUE TITLE
Classifier tasks should use valid, accessible fieldsets and legends

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/front-end-monorepo#2578]
## Package
lib-classifier

## Describe the bug
Classifier tasks use [`<fieldset>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) to wrap a task, with [`<legend>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/legend) to mark up the task instruction. `legend` can accept short, single strings (phrasing content) so this is great for question tasks or grouped text boxes and dropdown menus. The legend is prepended to the labels of the fieldset's inputs, in the accessibility tree. In this example code, a visually impaired user would hear 'Personal details: email' and 'Personal details: password' read out by their screenreader.

```html
<fieldset>
  <legend>Personal details</legend>
  <label for="email">E-mail</label>
  <input id="email" type="text" />
  <label for="password">Password</label>
  <input id="password" type="password" />
</fieldset>
```

Some projects, such as Beyond Borders, are entering paragraphs of formatted text as the task instruction in the project builder. Multiple paragraphs are invalid as legend text (short, single strings.) Invalid HTML may render unexpectedly in browsers, and will be confusing in screenreaders.

For example, on Beyond Borders, the transcription task label, in the accessibility tree, is:
```
Welcome to the new Transcription Task! There are two ways to transcribe, depending on if a volunteer has seen an image previously. 1. If the document has no previous volunteer-made marks: • Click under the beginning of the text line and at the end of the text line--this will create a green line under the text. Please make your marks in the order you’re reading the text. • DO NOT draw a second line if a line of text already has a purple line, even if the line is poorly drawn. • You may want to start your line a little before the first word to avoid obscuring text with the black \"x\" (line delete button). • A transcription box will pop-up with instructions. 2. If the document has previous transcription: • Purple lines will already be visible one the page. • Click on a purple line to view, select, edit, and/or submit previous transcriptions. • If there is additional text that has not been underlined—please see step one and make the first transcription. DO NOT draw a second line if a purple line is already associated with a line of text, even if the line is poorly drawn. Once a line is “retired” it will turn GRAY. • If all lines on a page have turned gray, you may retire the page by clicking “yes” to the question below—this will retire the entire page. • You may notice some Rogue Lines that are either not associated with a line of text, are a duplicate line for the same line of text or that are associated with archival marks that don’t need to be transcribed. If all lines except these are gray on a page, please click “yes” to indicate all lines are gray. For more detailed information see Line Agreement, Multiple/Rogue Lines and Retirement in the field guide • Lines will turn gray when three volunteers have transcribed the line exactly the same way. Transcribe the line the way YOU see it . If your version differs slightly in terms of spelling, capitalization or another factor, please add a new transcription, even if that means needing more transcribers to retire the line. For more detailed information and how to handle multiple dots and dashes , see Line Agreement and Retirement in the field guide. You can now work by the line! No need to transcribe an entire page to contribute your transcriptions. While you can save each line and submit your work anytime, you cannot return to a page after clicking done or refresh. Transcribe ALL text--both printed and handwritten.If a page is too difficult to decipher, click your browser's refresh button and move on to the next page .
```

## To Reproduce
1. Go to https://www.zooniverse.org/projects/mainehistory/beyond-borders-transcribing-historic-maine-land-documents/classify/workflow/18383
2. Right click on the task instruction and inspect it in the browser's accessibility tree.

## Expected behavior
Task instructions should be short, single strings of text.

## Additional context
Russ Wheatley has compiled some useful tests of [fieldset legends in various screenreader and browser combinations](https://russmaxdesign.github.io/Accessibility-test-cases/05-forms/fieldset.html).

**Comment by snblickhan** [comment]
The issue here is that there's a tension between seeing this field as a 'label' for the data being produced through the task vs. seeing it as a space for task instructions. 

We know from experience that if instructions are ONLY in a tutorial, or in the Tutorial + Help text, and NOT in the task area, it increases the likelihood of volunteers misinterpreting the task. It's SO much easier for volunteers if the instruction for that task is visible without need of an additional click (back into the Tutorial, or into the Help area). This is especially true of workflows that include different task types.

I think this is a difference in use that has been a longer shift in terms of UX design in our platform (particularly with the rise of text transcription projects). It feels like this approach (task area as label instead of instruction) might be over-fitting to NFN-style projects which are actually aiming to collect labels from a form, and for which the task will be the same each time, just applied to different areas on a page. The transcription task auto-label, which you've included here as an example of 'paragraph' text, is actually the default instruction for that task type, the team didn't write it. It was created with input from the Zooniverse FE team, is critical to successful use of the tool.

So the question becomes: how do we separate task instruction from a task label without requiring an additional click? The answer here isn't simply to restrict volunteer options. We need to think about how the tools are actually used before we apply rules across the board.

**Comment by eatyourgreens** [comment]
That doesn’t change the fact that the current markup makes the task instruction inaccessible to a screen reader. The whole field is concatenated into a single, long string. 

The GDS guidance for text inputs has good advice for building accessible forms, backed by research with disabled users. 

>Do not use long paragraphs and lists in hint text. Screen readers read out the entire text when users interact with the form element. This could frustrate users if the text is long.

**Comment by eatyourgreens** [comment]
What I'm saying is, if you want to add instructions to forms, you need to do it with valid, semantic HTML. You can't just cram a bunch of paragraphs into the existing instruction field.

**Comment by eatyourgreens** [comment]
Here's that same task, but written in valid HTML. The long description is moved outside the form. The fieldset and radio button have valid labels. This version should work in a screen reader, or with dictation software like Dragon. For example, someone with RSI should be able to use this form without having to awkwardly (or painfully) click with a mouse. They can select the radio inputs using dictation instead.

There's a secondary UX issue, which is that the transcription task uses a radio button group with only one radio button, but that's a separate problem. The idea here was that we would support more than one drawing tool for transcription workflows.

```html
<p><strong>Welcome to the new Transcription Task!</strong></p>
<p>There are two ways to transcribe, depending on if a volunteer has seen an image previously.</p>
<p><strong>1. If the document has no previous volunteer-made marks:</strong></p>
<ul>
<li>
<p>Click under the beginning of the text line and at the end of the text line--this will create a green line under the text. Please make your marks in the order you’re reading the text.</p>
</li>
<li>
<p><strong>DO NOT</strong> draw a second line if a line of text already has a purple line, <strong>even if</strong> the line is poorly drawn.</p>
</li>
<li>
<p>You may want to start your line a little before the first word to avoid obscuring text with the black "x" (line delete button).</p>
</li>
<li>
<p>A transcription box will pop-up with instructions.</p>
</li>
</ul>
<p><strong>2. If the document has previous transcription:</strong></p>
<ul>
<li>
<p>Purple lines will already be visible one the page.</p>
</li>
<li>
<p>Click on a purple line to view, select, edit, and/or submit previous transcriptions.</p>
</li>
<li>
<p>If there is additional text that has not been underlined—please see step one and make the first transcription. <strong>DO NOT</strong> draw a second line if a purple line is already associated with a line of text, even if the line is poorly drawn.</p>
</li>
</ul>
<p><strong>Once a line is “retired” it will turn GRAY.</strong></p>
<ul>
<li>
<p>If all lines on a page have turned gray, you may retire the page by clicking “yes” to the question below—this will retire the entire page.</p>
</li>
<li>
<p>You may notice some <strong>Rogue Lines</strong> that are either not associated with a line of text, are a duplicate line for the same line of text or that are associated with archival marks that don’t need to be transcribed. <strong>If all lines except these are gray on a page, please click “yes”</strong> to indicate all lines are gray. For more detailed information see <strong>Line Agreement, Multiple/Rogue Lines and Retirement</strong> in the field guide</p>
</li>
<li>
<p>Lines will turn gray when three volunteers have transcribed the line exactly the same way. <strong>Transcribe the line the way YOU see it</strong>. If your version differs slightly in terms of spelling, capitalization or another factor, please add a new transcription, even if that means needing more transcribers to retire the line. For more detailed information and <strong>how to handle multiple dots and dashes</strong>, see <strong>Line Agreement and Retirement</strong> in the field guide.</p>
</li>
</ul>
<p><strong>You can now work by the line!</strong> No need to transcribe an entire page to contribute your transcriptions. While you can save each line and submit your work anytime, you cannot return to a page after clicking done or refresh.</p>
<p class="StyledParagraph-sc-tbetod-0 iLSkwl"><strong>Transcribe ALL text--both printed and handwritten.</strong></p>
<p class="StyledParagraph-sc-tbetod-0 iLSkwl"><strong>If a page is too difficult to decipher, click your browser's refresh button and move on to the next page</strong>.</p>
<form>
  <fieldset>
    <legend>Select a tool</legend>
    <label>
      <img src='./assets/transcription-icon.svg' alt='Transcription drawing tool.'>
      Single line of text (0 drawn)
      <input type=radio name='drawing-tool' value='0'>
    </label>
  <fieldset>
  <fieldset>
    <legend>Have all the volunteer-made underline marks turned grey?</legend>
    <label>
      Yes
      <input type=radio name=T1 value='0'>
    </label>
    <label>
      No
      <input type=radio name=T1 value='1'>
    </label>
  </fieldset>
</form>
```

**Comment by eatyourgreens** [comment]
Considering an audience that has problems using a mouse or trackpad (eg. RSI or arthritis), there is still an awful lot of scrolling in order to get to the Yes/No question in that form.


https://github.com/zooniverse/front-end-monorepo/assets/59547/062dd013-4b00-4568-a0ed-6c1b69fe0cd4

**Comment by eatyourgreens** [comment]
NB. In my code example, I've moved the markdown text outside the fieldset, and form, as non-form elements would be skipped by both JAWS and NVDA in [forms/focus mode](https://www.accessibility-developer-guide.com/knowledge/screen-readers/desktop/browse-focus-modes/#associated-information).

**Comment by goplayoutside3** [comment]
I think in this scenario, two things are true at the same time: 
1. Transcription projects have a very involved subject viewer with various line colors and multiple steps needed to complete a transcription task. This makes the task "special" in that additional visible instructions helps first-time volunteers understand what's expected of them.
2. The "Main Text" field in the workflow editor part of the Project Builder is used for the html form label in the task area, and must follow web accessibility rules. The paragraphs of text currently used for a transcription task's "Main Text" is not accessible to screen readers, requires volunteers to scroll down to answer the question, and should be shortened to a simple, plain text label.

In order to address both points, I propose we make some modifications to the task area's "Need some help with this task?" modal. The modal currently behaves as the tutorial modal does, it's centered and must be closed in order to view or interact with the classifier interface. In contrast, the field guide's modal is intentionally movable and does not cover the subject image. Also related to the field guide modal, Sam has requested in the past (Issue https://github.com/zooniverse/front-end-monorepo/issues/2136) for the field guide modal to stay open when a volunteers clicks outside of it, so the field guide can be referenced even while interacting with the classifier. I think we could give the same treatment to the "Need some help with this task?" modal - make it movable and stays open until closed via escape key or close button.

With a movable modal easily viewed at the same time as the classifier interface, first-time volunteers who need help understanding the transcription task won't need the paragraphs of text in the task area. Detailed instructions can instead be placed in the modal. @snblickhan please confirm this sounds like a good solution, and I'll create a separate Issue focused on modal modification. I will also re-open #2136 if desired.

**Comment by snblickhan** [comment]
This is a great solution. As part of this implementation, we can update the pre-filled text for the Transcription Task to be a short instruction on how to use the tool, and then project builders can include much more detailed instruction in the Help Text.

**Comment by eatyourgreens** [comment]
This got derailed into a discussion of the instructions for transcription tasks, but other tasks also have invalid labels and legends (drop-down tasks, survey tasks, drawing tasks etc.)

**Comment by eatyourgreens** [comment]
The drop-down task has a legend where it should have a label. I'm not sure how that slipped through code review. 

https://github.com/zooniverse/front-end-monorepo/blob/45dde3a406671d480e97142789375d595afa086a/packages/lib-classifier/src/plugins/tasks/dropdown-simple/components/SimpleDropdownTask.js#L72-L76

**Comment by eatyourgreens** [comment]
I take that back about the survey task. It has valid legends, because it renders invalid content as inline text. 

https://github.com/zooniverse/front-end-monorepo/blob/45dde3a406671d480e97142789375d595afa086a/packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/Chooser.js#L23-L33

**Comment by eatyourgreens** [comment]
I’ll also leave this quote, from a volunteer who is unable to use a mouse or trackpad and dictates their survey task classifications with Dragon Naturally Speaking, to demonstrate why all forms should use valid, labelled inputs. 

>“The browser I use is Google Chrome and just this morning I was working with some of your ideas and discovered that I could actually say the words and it would click on them. Example: play, deer, nothing there, bison, etc. This is the best of the best so it's now very usable. Feel free to pass this on, if there's others similar to myself.”
Owen (Dragon Naturally Speaking)

**Comment by eatyourgreens** [comment]
Re. the transcription task, this isn't a comprehensive audit by any means, but the following WCAG success criteria are level A failures.

- [SC 1.4.1: Use of colour (level A)](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html): the task uses colour to convey whether a line is complete or not.
- [SC 2.1.1: Keyboard (level A.)](https://www.w3.org/WAI/WCAG21/Understanding/keyboard): Existing lines can be edited from the keyboard, but lines can't be created or moved from the keyboard (see https://github.com/zooniverse/front-end-monorepo/discussions/2422.)

**Comment by goplayoutside3** [comment]
> The drop-down task has a legend where it should have a label for the select menu. I'm not sure how that slipped through code review.

Please feel free to open a PR to fix.
